crates/bevy_time/src/fixed_timestep.rs

```diff
@@ -1,321 +1,162 @@
+//! two times during a ~16.667ms frame, once during a ~8.333ms frame, and once every two frames
+//! with ~4.167ms frames. However, the same criteria may not result in exactly 8.333ms passing
+//! between each execution.
+//!
+//! When using fixed time steps, it is advised not to rely on [`Time::delta`] or any of it's
+//! variants for game simulation, but rather use the value of [`FixedUpdate`] instead.
 
-/// A system run criteria that enables systems or stages to run at a fixed timestep between executions.
-///
-/// This does not guarantee that the time elapsed between executions is exactly the provided
-/// fixed timestep, but will guarantee that the execution will run multiple times per game tick
-/// until the number of repetitions is as expected.
-///
-/// For example, a system with a fixed timestep run criteria of 120 times per second will run
-/// two times during a ~16.667ms frame, once during a ~8.333ms frame, and once every two frames
-/// with ~4.167ms frames. However, the same criteria may not result in exactly 8.333ms passing
-/// between each execution.
-///
-/// When using this run criteria, it is advised not to rely on [`Time::delta`] or any of it's
-/// variants for game simulation, but rather use the constant time delta used to initialize the
-/// [`FixedTimestep`] instead.
-///
-/// For more fine tuned information about the execution status of a given fixed timestep,
-/// use the [`FixedTimesteps`] resource.
-pub struct FixedTimestep {
-    state: LocalFixedTimestepState,
-    internal_system: Box<dyn System<In = (), Out = ShouldRun>>,
+use crate::Time;
+use bevy_app::CoreSchedule;
+use bevy_ecs::{system::Resource, world::World};
+use bevy_utils::Duration;
+use thiserror::Error;
+
+/// The amount of time that must pass before the fixed timstep schedule is run again.
+#[derive(Resource, Debug)]
+pub struct FixedTime {
+    accumulated: Duration,
+    /// Defaults to 1/60th of a second.
+    /// To configure this value, simply mutate or overwrite this resource.
+    pub period: Duration,
 }
 
-impl Default for FixedTimestep {
-    fn default() -> Self {
-        Self {
-            state: LocalFixedTimestepState::default(),
-            internal_system: Box::new(IntoSystem::into_system(Self::prepare_system(
-                Default::default(),
-            ))),
+impl FixedTime {
+    /// Creates a new [`FixedTime`] struct
+    pub fn new(period: Duration) -> Self {
+        FixedTime {
+            accumulated: Duration::ZERO,
+            period,
         }
     }
-}
 
-impl FixedTimestep {
-    /// Creates a [`FixedTimestep`] that ticks once every `step` seconds.
-    pub fn step(step: f64) -> Self {
-        Self {
-            state: LocalFixedTimestepState {
-                step,
-                ..Default::default()
-            },
-            ..Default::default()
+    /// Creates a new [`FixedTime`] struct with a period specified in `f32` seconds
+    pub fn new_from_secs(period: f32) -> Self {
```

Comment:
should there also be a `new_from_duration` constructor for users who want to use other units?

Reply:
IIRC `new` does this. We'll want to improve the whole API for this in a follow-up though: I tried to stick to something minimal for reviewability.